ros/lib/rbac_interface.py

```diff
@@ -0,0 +1,138 @@
+from urllib.parse import urljoin
+from http import HTTPStatus
+from flask_restful import abort
+from .config import RBAC_SVC_URL, PATH_PREFIX
+import requests
+
+
+RBAC_SVC_ENDPOINT = "/api/rbac/v1/access/?application=%s"
+AUTH_HEADER_NAME = "X-RH-IDENTITY"
+VALID_HTTP_VERBS = ["get", "options", "head", "post", "put", "patch", "delete"]
+
+
+def fetch_url(url, auth_header, logger, method="get"):
+    """
+    Helper to make a single request *** Add to this comment once functionality done.
+    """
+    if method not in VALID_HTTP_VERBS:
+        abort(
+            HTTPStatus.METHOD_NOT_ALLOWED, message="'%s' is not valid HTTP method." % method
+        )
+
+    logger.debug("Fetching %s" % url)
+
+    response = requests.request(method, url, headers=auth_header)
+    logger.debug("fetched %s" % url)
+    _validate_service_response(response, logger, auth_header)
+    return response.json()
+
+
+def _validate_service_response(response, logger, auth_header):
+    """
+    Raise an exception if the response was not what we expected.
+    """
+    print(response.status_code)
+    if response.status_code == requests.codes.not_found:
+        abort(
+            response.status_code, message="%s error received from service: %s" % response.text
```

Comment:
Will this line execute without an error?

Reply:
What does the `abort` call do?

Reply:
@beav The abort helper will wrap an HTTP Exception with the status and message provided in the method call. https://flask-restful.readthedocs.io/en/latest/api.html#flask_restful.abort
@kgaikwad Do you mean will this line execute if there is no error? Also, I have made changes in the structure of if-else here.

Reply:
> @kgaikwad Do you mean will this line execute if there is no error? Also, I have made changes in the structure of if-else here.

@apuntamb, I meant when there is 404 status code, will this below line run?
 `abort(response.status_code, message="%s error received from service: %s" % response.text)` 
 
You already modified the message so it won't be an issue now. 
But for your understanding, could you please run below statement into python shell? 
`"%s error received from service: %s" % response.text`

As per my understanding it should error out as you have passed a single argument to message string.

Reply:
Ahh okk, I got your concern now. Yes there were two formatters `%s` added and only one value passed, so yes it would error out. I had made certain changes post testing so, mistakenly forgot to correct the statement. :)
I have corrected the same in latest commit.